cmake/DefaultTargetProperties.cmake

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(set_default_target_properties target_name)
+    #-------------------------------------------------------------------
+    # C++ standard
+
+    # Minimum C++ standard
+    target_compile_features(${target_name} PRIVATE cxx_std_14)
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Ensure the proposed compiler supports our minimum C++
+        # standard.
+        CXX_STANDARD_REQUIRED ON
+        # Disable compiler extensions. E.g. use -std=c++11  instead of
+        # -std=gnu++11.  Helps limit cross-platform issues.
+        CXX_EXTENSIONS OFF
+    )
+
+    #-------------------------------------------------------------------
+    # Compiler warnings
+
+    include(CompilerWarnings)
+    set_default_compiler_warnings(${target_name})
+
+
+    #-------------------------------------------------------------------
+    # Potentially enable sanitizers.
+    include(Sanitizers)
+    enable_sanitizers(project_options)
+
+
+    #-------------------------------------------------------------------
+    # Symbol visibility
+
+    # Hide symbols from this library by default.
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        C_VISIBILITY_PRESET hidden
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN YES
+    )
+
+    # Hide all symbols from external statically linked libraries.
+    if (CMAKE_CXX_COMPILER_ID MATCHES ".*Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_link_options(${target_name} PRIVATE -Wl,--exclude-libs,ALL)
+    endif ()
+
+    # Use C++11 ABI by default, but allow disabling.
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
+        CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 5.0)
+        option(GLIBCXX_USE_CXX11_ABI "For gcc, use the new C++11 library ABI" ON)
```

Comment:
Heads up that in vfxref specs (at least 20, 21 and 22) this is _disabled_.

See the notes on GCC6 and 9 on https://vfxplatform.com/.

Reply:
So this should be `OFF` by default as per current VFX reference?

Its frustrating that the default depends on OS as well as gcc version, and of course must be set to match the dependencies you're linking.

Reply:
Yes, off by default. Let the user decide of course, and you have the option, but should probably follow the guidance that the VFX community follows.

Reply:
Disabled by default in 9504f44b (later refactored in 1c32585a)